Task16 xml

andresmadeddie opened the pull request:
Create XML, XDS, and validate/parser files

andresmadeddie committed:
Added DOMParser, XML Validator, shop.xml and shop.xsd

andresmadeddie committed:
Added DOMParser, XML Validator, shop.xml and shop.xsd

andresmadeddie committed:
Added DOMParser, XML Validator, shop.xml and shop.xsd

andresmadeddie committed:
fusion XML Parse and Validate files. modify xml and xsd files

ankochergin commented on src/main/java/com/solvd/util/XMLValidatorAndParser.java:
```diff
@@ -0,0 +1,75 @@
+import javax.xml.validation.Validator;
+import java.io.File;
+
+public class XMLValidatorAndParser {
```
split into 2 classes (1 for util class with parser and 1 util class for XML validator)
also you can call these util classes from main

ankochergin commented on src/main/java/com/solvd/util/XMLValidatorAndParser.java:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    public static void DOMParser(String xmlFilePath) {
```
don't forget about naming conventions

andresmadeddie committed:
Separate validator and parser files. Use both new classes to validate and parse in the main method

ankochergin commented on src/main/java/com/solvd/util/XmlParser.java:
```diff
@@ -0,0 +1,38 @@
+import java.io.File;
+
+public class XmlParser {
+    public static void Parser(String xmlFilePath) {
```
```suggestion
    public static void parse(String xmlFilePath) {
```
method names must start with lowercase letter

ankochergin commented on src/main/java/com/solvd/util/XmlValidator.java:
```diff
@@ -0,0 +1,41 @@
+            e.printStackTrace();
+        }
+        return "XML is not valid against the XSD";
+    }
```
```suggestion
  
```

ankochergin commented on src/main/java/com/solvd/Main.java:
```diff
@@ -342,5 +342,7 @@ public static void main(String[] args) {
         System.out.println(productPromotionService.getAll());
 
+        // XML Validate and Parse
+        XmlParser.Parser(XmlValidator.Validator());
```
```suggestion
        XmlValidator.validateXMLAgainstXSD("src/main/resources/shop.xml", "src/main/resources/shop.xsd");
        XmlParser.parse("src/main/resources/shop.xml");
```

andresmadeddie committed:
Change methods names to parser and validator
Modify main method to validate and parses in different lines

andresmadeddie committed:
Change methods names to parser and validator
Modify main method to validate and parses in different lines and modify XmlValidator class accordingly

andresmadeddie committed:
Change methods names to parser and validator
Modify main method to validate and parses in different lines and modify XmlValidator class accordingly

andresmadeddie committed:
Change methods names to parser and validator
Modify main method to validate and parses in different lines and modify XmlValidator class accordingly
